sky/utils/dag_utils.py

```diff
@@ -11,6 +12,52 @@
 
 logger = sky_logging.init_logger(__name__)
 
+# Message thrown when APIs sky.{exec,launch,spot_launch}() received a string
+# instead of a Dag.  CLI (cli.py) is implemented by us so should not trigger
+# this.
+_ENTRYPOINT_STRING_AS_DAG_MESSAGE = """\
+Expected a sky.Task or sky.Dag but received a string.
+
+If you meant to run a command, make it a Task's run command:
+
+    task = sky.Task(run=command)
+
+The command can then be run as:
+
+  sky.exec(task, cluster_name=..., ...)
+  # Or use {'V100': 1}, 'V100:0.5', etc.
+  task.set_resources(sky.Resources(accelerators='V100:1'))
+  sky.exec(task, cluster_name=..., ...)
+
+  sky.launch(task, ...)
+
+  sky.spot_launch(task, ...)
+""".strip()
+
+
+def convert_entrypoint_to_dag(entrypoint: Any) -> 'dag_lib.Dag':
+    """Convert the entrypoint to a sky.Dag.
+
+    Raises TypeError if 'entrypoint' is not a 'sky.Task' or 'sky.Dag'.
+    """
+    # Not suppressing stacktrace: when calling this via API user may want to
+    # see their own program in the stacktrace. Our CLI impl would not trigger
+    # these errors.
+    if isinstance(entrypoint, str):
+        raise TypeError(_ENTRYPOINT_STRING_AS_DAG_MESSAGE)
```

Comment:
`with ux_utils.print_exception_no_traceback`; ditto for following raises

Reply:
Good catch! Updated. Thanks!